frontend/src/pages/article/components/article-form/article-form.tsx

```diff
@@ -49,22 +50,33 @@ const ArticleForm: React.FC<Properties> = ({ articleForUpdate }) => {
   const handleArticleSubmit = useCallback(
     (articleSubmitType: ValueOf<typeof ArticleSubmitType>) =>
       (payload: ArticleRequestDto): void => {
+        const isArticlePublished =
+          articleSubmitType === ArticleSubmitType.PUBLISH;
+
         const updatedPayload = {
           ...payload,
-          publishedAt:
-            articleSubmitType === ArticleSubmitType.PUBLISH
-              ? new Date().toISOString()
-              : null,
+          publishedAt: isArticlePublished ? new Date().toISOString() : null,
         };
 
         void dispatch(
           articlesActions.createArticle({
             articlePayload: updatedPayload,
             generatedPrompt: getGeneratedPromptPayload(generatedPrompt),
           }),
-        );
+        )
+          .unwrap()
+          .then(() => {
+            navigate(
+              isArticlePublished
+                ? AppRoute.ARTICLES
+                : ArticleSubRoute.MY_ARTICLES,
+            );
+
+            dispatch(promptActions.resetPrompts());
```

Comment:
If you need to reset prompts on component exit - then move it to useEffect cleanup